test/unit/deferred.js

```diff
@@ -525,6 +525,34 @@ QUnit.test( "jQuery.Deferred.then - spec compatibility", function( assert ) {
 	} catch ( _ ) {}
 } );
 
+QUnit[ window.console ? "test" : "skip" ]( "jQuery.Deferred.exceptionHook", function( assert ) {
+
+	assert.expect( 1 );
+
+	var done = assert.async(),
+		defer = jQuery.Deferred(),
+		oldWarn = window.console.warn;
+
+	window.console.warn = function( msg ) {
```

Comment:
> Agreed.

Yeah, but if we would use it across all the code base i.e. everywhere, it seems as matter of consistency and usability for other not that simple cases like this one.

>  A few years ago I tried it and saw a bloat library polluting the global scope and QUnit's api. 

Is it possible to provide an extension to qunit as separate package? Since our extensions already complicate our test code, i fear core testsuite might became more vague if we would extend it even more.